### ISSUE TITLE
Sample code for "Optimizing an indicator parameter" doesn't work by default

### ISSUE THREAD

**Issue by vaidab** [freqtrade/freqtrade#7088]
There are 3 issues when trying to run the code:

### 1. Error in running
If you copy/paste the code here: https://www.freqtrade.io/en/stable/hyperopt/#optimizing-an-indicator-parameter
And run it, you get:
"Failed validating 'minProperties' in schema['properties']['minimal_roi']:
    {'minProperties': 1,
     'patternProperties': {'^[0-9.]+$': {'type': 'number'}},
     'type': 'object'}

On instance['minimal_roi']:
    {}
jsonschema.exceptions.ValidationError: {} does not have enough properties"

This can be fixed by adding minimal_roi to the strategy.

### 2. Performance warning
Then it gives a performance warning (which should be explained in the doc):
"PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`"

### 3. User Warning
There are a lot of warnings like this one:

.local/lib/python3.10/site-packages/skopt/optimizer/optimizer.py:449: UserWarning: The objective has been evaluated at this point before.
  warnings.warn("The objective has been evaluated "

They could be there because of the parameters:
    buy_ema_short = IntParameter(3, 50, default=5)
    buy_ema_long = IntParameter(15, 200, default=50)

It seems: 15-50 are common parameters to both long and short.

**Comment by xmatthias** [comment]
Point 2 is explained in the [advanced strategy](https://www.freqtrade.io/en/latest/strategy-advanced/#performance-warning) section - and can be rewritten in a way to avoid the warning.

That said - the `.copy()` is done implicitly after the populate-indicators in hyperopt - so it's usually not really an issue - which i've now also added ot the docs.

I've now adapted the documentation a bit to explain the points raised more clearly.

---

We're also more than happy to get contributions to the documentation, explaining things better / in simpler terms. 
As developers, it's often difficult to properly write documentation a user without prior experience can understand, as such it's often best to get contributions for this, as wording can then often be improved.
